### PR TITLE
[MB-15120] PPM Processor can confirm approval/rejection of PPM Document Sets

### PR DESCRIPTION
**Pull request by bellex3**
## Summary
Integration ticket Service Counselor's ability to confirm the approval/rejection of PPM document sets. 
1) Created 3 Storybook components for the PPM shipment card when the SC approves, rejects, and excludes items. All 3 will display the "new packet ready for download" label.

- [Approved](https://output.circle-artifacts.com/output/job/135dd74b-2bbf-4d83-a7a1-d33fae51f21f/artifacts/0/storybook/index.html?path=/story/office-components-shipment-display--ppm-shipment-service-counselor-approved)
- [Rejected](https://output.circle-artifacts.com/output/job/135dd74b-2bbf-4d83-a7a1-d33fae51f21f/artifacts/0/storybook/index.html?path=/story/office-components-shipment-display--ppm-shipment-service-counselor-rejected)
- [Excluded](https://output.circle-artifacts.com/output/job/135dd74b-2bbf-4d83-a7a1-d33fae51f21f/artifacts/0/storybook/index.html?path=/story/office-components-shipment-display--ppm-shipment-service-counselor-excluded)

2) Created tests in ShipmentDisplay (test for packet download label) & ReviewDocuments (test for post documents status function)
3) Created useMutation function and handleSubmit function in ReviewDocumentsSidePanel to hook up the handler for submitting documents review to the frontend. 

**IMPORTANT PLEASE NOTE:**
I paired with @scottmries on this ticket since his [ticket MB-15105](https://dp3.atlassian.net/browse/MB-15105?atlOrigin=eyJpIjoiMmY4MjgyMDViZWEyNDQ5YjlhYjY1NTk1ODhiOWIyOTgiLCJwIjoiaiJ9) deals with the backend code for the handler. Currently, if you try to test out the "Confirm" button functionality on the browser, you will get an error. This is normal right now because this branch does not contain Scott's backend code for the functionality to work. 

The plan is to merge in Scott's code once his PR is approved and then we will be able to fully test out the Confirm button functionality for this ticket. 

For right now, **Design** could check out the [Storybook components](https://output.circle-artifacts.com/output/job/135dd74b-2bbf-4d83-a7a1-d33fae51f21f/artifacts/0/storybook/index.html?path=/story/office-components-shipment-display--ppm-shipment-service-counselor-approved) to see the label. **Engineers** review the PR based on the code. 
## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Go to http://officelocal:3000/
2. Click Local sign-in & log in as an existing user called 'services_counselor_role@office.mil (PPM office)'
3. At the top click "PPM Closeout" (under 'Moves' header).
4. Select 'ei5jhzbd, rhruocustomer' (HG3RCB). Should be the 1st row.
5. Click 'Review Documents'.
6. Click 'Continue' button 2 times (1st doc is weight ticket, 2nd doc is pro-gear) to get to the last Confirmation (Send to customer?) panel. 
7. Click the "Confirm" button.

**EXPECTED FUNCTIONALITY**
8. It should take you back to the Move Details page and on the PPM Shipment card, you should see a gray label at the heading saying "PACKET READY FOR DOWLOAD".

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/main/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

## Screenshots
Screenshot 2023-04-11 at 10 59 38 AM

[MB-15120]: https://dp3.atlassian.net/browse/MB-15120?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by bellex3** [comment]
> @bellex3 I looked at the storybook component and it looks good to me. I checked [the Happo diffs](https://happo.io/a/488/p/603/compare/6513b22d8f89e6249b548afac6ab4a853e47f8e3/07e79845d9d035dcce7d1b5047c0fbce4e8d5efe) and the first diff is throwing an error. I'm not sure if it's related to this PR, but wanted to call it out just in case!

@rileyorr thank you for catching that. I had to fix some code in the Storybook component for ReviewDocumentsSidePanel. Here are the new [Happo diffs](https://happo.io/a/488/p/603/compare/6513b22d8f89e6249b548afac6ab4a853e47f8e3/363b2c5a3a268860c825f5e7e80c101632316d1a) with no errors.

**Comment by rileyorr** [comment]
@bellex3 thanks for that update! I just took another look and noticed that the 'After' section in the side by side doesn't have the status tag. It looks correct in Storybook, so it might just be something strange going on with Happo diffs. Just wanted to double-check because I'm not able to test that the label pops up in the ephemeral deploy since the backend isn't hooked up yet. 

Screenshot for reference:

Screenshot 2023-04-14 at 9 28 37 AM

File: src/components/Office/ShipmentDisplay/ShipmentDisplay.test.jsx [hunk 1]
```diff
@@ -222,38 +222,6 @@ describe('Shipment Container', () => {
...
-              ordersLOA={ordersLOA}
-              shipmentType={SHIPMENT_OPTIONS.PPM}
-              isSubmitted
-              allowApproval={false}
-              warnIfMissing={['counselorRemarks']}
-              reviewURL="/"
-            />
-          </MockProviders>,
-        );
-        expect(screen.getByTestId('tag', { name: 'packet ready for download' })).toBeInTheDocument();
```
Comment (pearl-truss) [inline src/components/Office/ShipmentDisplay/ShipmentDisplay.test.jsx:239]: **Suggestion:** I think still having an assertion for the case when the packet is ready for download would be good here to test the logic that was changed in the component

**Review by pearl-truss — COMMENTED** [review]
Great job! I just left one small comment

File: src/components/Office/ShipmentDisplay/ShipmentDisplay.test.jsx (hunk 1 above)
Comment (bellex3) [reply src/components/Office/ShipmentDisplay/ShipmentDisplay.test.jsx:239]: @pearl-truss Thanks and makes sense. I added the tests for excluded and rejected shipments.

**Comment by bellex3** [comment]
> @bellex3 thanks for that update! I just took another look and noticed that the 'After' section in the side by side doesn't have the status tag. It looks correct in Storybook, so it might just be something strange going on with Happo diffs. Just wanted to double-check because I'm not able to test that the label pops up in the ephemeral deploy since the backend isn't hooked up yet.
> 
> Screenshot for reference:
> 
> Screenshot 2023-04-14 at 9 28 37 AM

@rileyorr I refined the previous ShipmentDisplay Storybook component and now have 3 different Storybook components that each show the Packet Ready for Download label for when a PPM shipment is 
1) [approved](https://output.circle-artifacts.com/output/job/135dd74b-2bbf-4d83-a7a1-d33fae51f21f/artifacts/0/storybook/index.html?path=/story/office-components-shipment-display--ppm-shipment-service-counselor-approved),
2) [rejected](https://output.circle-artifacts.com/output/job/135dd74b-2bbf-4d83-a7a1-d33fae51f21f/artifacts/0/storybook/index.html?path=/story/office-components-shipment-display--ppm-shipment-service-counselor-rejected),
3) and if items were [excluded](https://output.circle-artifacts.com/output/job/135dd74b-2bbf-4d83-a7a1-d33fae51f21f/artifacts/0/storybook/index.html?path=/story/office-components-shipment-display--ppm-shipment-service-counselor-excluded). 
 
I felt this would be more clear so people are aware that the label shows up for these statuses. 

Also, I fixed the Happo issue and now there is only one [Diff ](https://happo.io/a/488/p/603/compare/fada06f2dffb67ce0ea77c623d9d987918ef82ed/f692f809fd99b1d307062c02ea1f55febcdc5e73)and the label shows for After now.

**Review by pearl-truss — APPROVED** [review]
Thanks for making that change! The code looks good to me :)

**Review by rileyorr — APPROVED** [review]
@bellex3 thank you for providing that clarification! Everything looks good to me.

**Pull request merged**